docs/02-app/02-api-reference/01-directives/use-cache.mdx

```diff
@@ -63,324 +55,227 @@ export async function getData() {
 }
 ```
 
-> **Good to know**: Functions that use the `use cache` directive must not have any side-effects, such as modifying state, directly manipulating the DOM, or setting timers to execute code at intervals
+## Good to know
 
-## Revalidating
+- `use cache` is an experimental Next.js feature, and not a native React feature like [`use client`](/docs/app/api-reference/directives/use-client) or [`use server`](/docs/app/api-reference/directives/use-server).
+- The arguments (or props) passed to a cached function must be [serializable](https://react.dev/reference/rsc/use-server#serializable-parameters-and-return-values) by React. This means they can be converted to a format like JSON. Non-serializable props like functions or React elements cannot be used as arguments.
```

Comment:
>Non-serializable props like functions or React elements cannot be used as arguments.

I don't think this is correct. AFAIK they _can_ be used as arguments (it's [how composition works](https://bsky.app/profile/danabra.mov/post/3l7dlqdamr52j)) but they'll be treated as opaque — the only thing you can do with them is to embed them in the output / pass them down to the client. You can't introspect (for elements) or call (for functions).

Reply:
https://github.com/vercel/next.js/pull/72286